test/compflags/ferguson/default-binary-name/sub_test

```diff
@@ -0,0 +1,62 @@
+#! /usr/bin/env python
+
+import subprocess
+import sys
+import os
+
+if len(sys.argv)!=2:
+  print 'usage: sub_test COMPILER'
+  sys.exit(0)
+
+# Find the base installation
+compiler=sys.argv[1]
+if not os.access(compiler,os.R_OK|os.X_OK):
+  Fatal('Cannot execute compiler \''+compiler+'\'')
+
+# find current directory
+
+def run(source, exe_name, compopts, expect): 
+  p = subprocess.Popen([compiler, source] + compopts,
+                       stdout=subprocess.PIPE)
+  myoutput = p.communicate()[0]
+  sys.stdout.write(myoutput)
+  if p.returncode != 0:
+    sys.stdout.write("[Error matching compiler output for %s (%s)]\n"%(source, exe_name))
+    sys.exit(1)
+  else:
+    sys.stdout.write("[Success compiling %s (%s)]\n"%(source, exe_name))
+
+  returncode = -1
+  myoutput = ""
+
+  if os.access(exe_name,os.R_OK|os.X_OK):
+
+    numlocales = "0"
+    if "NUMLOCALES" in os.environ:
+      numlocales = os.environ['NUMLOCALES']
+
+    if numlocales == "0":
+      p = subprocess.Popen(["./%s" % exe_name], stdout=subprocess.PIPE)
+    else:
+      p = subprocess.Popen(["./%s" % exe_name, "-nl%s" % numlocales],
+          stdout=subprocess.PIPE)
+
+    myoutput = p.communicate()[0]
+    returncode = p.returncode
+
+  myoutput = myoutput.strip()
+  if returncode == 0 and myoutput == expect:
+    sys.stdout.write("[Success matching program output for %s (%s)]\n"%(source, exe_name))
+  else:
+    sys.stdout.write("[Error matching program output for %s (%s)]\n"%(source, exe_name))
+
+  if os.access(exe_name, os.F_OK):
+    os.remove(exe_name)
+  if os.access(exe_name + "_real", os.F_OK):
+    os.remove(exe_name + "_real")
+
+
+run("test-default-binary-name.chpl", "test-default-binary-name", [], "Hello")
+run("test-default-binary-name.chpl", "test_exe", ["-o", "test_exe"], "Hello")
```

Comment:
Could you add another test that sets CHPL_EXE_NAME to "test_exe" to show that setting it through the environment variable still works as expected?